### DISCUSSION TITLE
AttributeError: object has no attribute 'apply_async'

### DISCUSSION THREAD

**Discussion by Ahtii** [celery/celery#7319] [Q&A]
Hi,

We have an old project that we are upgrading to latest versions. The project has celery 3.1 we upgraded it to 5.2.3 and changed class based tasks in our codebase from this:

tasks.py

```
from celery import task

class MyTask(task.Task):
   def run(self, arg1, arg2, arg3):
      # do something 
```

and call it like this: `async_result = MyTask.delay(temp_obj, arg1, arg2)`

to this:

```
from celery import Task

class MyTask(Task):
   def run(self, arg1, arg2, arg3):
      # do something 
```

but when we call it like we were calling it before we get this error:

`AttributeError: 'Temp' object has no attribute 'apply_async'`

I tried look it up on docs didn't found anything. Please help

[a comment by an automated account is not shown]

**Comment by thedrow** [comment]
Please read the 4.0 and the 5.0 release notes first.

**Reply by Ahtii** [reply]
Okay solved it by changing to `MyTask().delay()` found it here https://docs.celeryproject.org/en/stable/internals/deprecation.html

**Reply by Ahtii** [reply]
Now I get another issue: `Received unregistered task of type None ` in celery output. I have registered the task but still I get the issue please help.

**Reply by thedrow** [reply]
I can't really help without all the information.
